plugins/source/mysql/resources/plugin/plugin_test.go

```diff
@@ -218,30 +166,55 @@ func TestPlugin(t *testing.T) {
 	}
 
 	// Init the plugin so we can call migrate
-	if err := p.Init(ctx, spec); err != nil {
+	if err := p.Init(ctx, specBytes); err != nil {
 		t.Fatal(err)
 	}
-	res := make(chan *schema.Resource, 1)
+	res := make(chan message.Message, 1)
 	g := errgroup.Group{}
 
 	g.Go(func() error {
 		defer close(res)
-		return p.Sync(ctx, syncTime, res)
+		opts := plugin.SyncOptions{
+			Tables:            []string{testTable.Name},
+			SkipTables:        []string{otherTable.Name},
+			DeterministicCQID: false,
+			StateBackend:      nil,
+		}
+		return p.Sync(ctx, opts, res)
 	})
-	resources := make([]*schema.Resource, 0)
+	actualRecords := make([]arrow.Record, 0)
 	for r := range res {
-		resources = append(resources, r)
+		insert := r.(*message.Insert).Record
+		actualRecords = append(actualRecords, insert)
 	}
 	err = g.Wait()
 	if err != nil {
 		t.Fatal("got unexpected error:", err)
 	}
-	if len(resources) != 2 {
-		t.Fatalf("expected 2 resource, got %d", len(resources))
+	if len(actualRecords) != 2 {
+		t.Fatalf("expected 2 resource, got %d", len(actualRecords))
 	}
-	expectedStrings := getExpectedStrings(t, testTable, data)
-	actualStrings := getActualStrings(resources)
-	sortResults(testTable, actualStrings)
 
-	require.Equal(t, expectedStrings, actualStrings)
+	sortResults(testTable, actualRecords)
+
+	for recordIndex, expectedRecord := range expectedRecords {
```

Comment:
Notice we switched from string comparison to value comparison as everything is arrow types now